### ISSUE TITLE
Feature Request: specify probabilities

### ISSUE THREAD

**Issue by jamesmf** [rodrigopivi/Chatito#48]
It would be great if you could specify ratios between entries or probabilities or relative counts. 

For instance if I have 

```
~[topping]
    anchovies
    pepperoni
```

I would love to be able to specify that anchovies are less common than pepperoni.
```
~[topping]
    anchovies &[1]
    pepperoni &[99]
```
Then for every 100 examples 99 would say `pepperoni` and only 1 would say `anchovies`

**Comment by rodrigopivi** [comment]
hi @jamesmf 
Thanks for your feedback, this is indeed a good addition to the DSL, i like your semantics of a new entity at the end of each sentence using &[1]. I think there are a couple of options here. e.g:

```
~[topping]
    pepperoni &[99]
    anchovies
    mushrooms
```

In this case then `anchovies` and `mushrooms` would share the remaining 1%?

Currently the way to handle probabilities is to just repeat the the same sentence or maybe add typos or other augmentation. e.g.:

```
~[topping]
    pepperoni
    pepperoni
    Pepperoni
    peperoni
    Peperoni
    anchovies
    mushrooms
```

So i agree in the sense that we may want to make this shorter, adding augmentation mechanisms and probabilities.

```
// augmentation can be: 'none', 'capitalization', 'typos' or 'full'
~[topping]('augmentation': 'full')
    pepperoni &[20x]
    anchovies 
    mushrooms
```

This would be another way to control probabilities, but instead of imperatively expressing the 99%, we can also express the frecuency, and it would be like if that word was 20 times listed, so i'm not sure which way is better atm. would love to hear any feedback

**Comment by jamesmf** [comment]
I think it would be reasonable if the & can be treated as a multiplier so that not much functionality would have to change. I don't think describing them as probabilities would work very well when you start dealing with multiple levels.

I just ran into the tricky situation where I had two very long lists `~[x]` and `~[y]` and then wanted to implement
```
@[capture]
    ~[x?] ~[y]
    ~[x] ~[y?]
```
Which off the top of my head I thought would have generated approx 50% of the data as either x or y, but not both. In reality, it was overwhelmingly likely to generate [x and y] because (I think) there were far, far, far more combinations of [x and y] than there were of [x or y]. 

If it is expanding all the combinations, then sampling some number of them, the combinatorial math gets really hard to actually estimate, and you end up with some seriously weird results (in the example above, I generated hundreds of examples and 0 showed up as [x or y]).
